capstone/src/main/java/com/google/step/clients/MetricsClient.java

```diff
@@ -177,6 +176,14 @@ public boolean putPageView(String restaurantName, String restaurantKey) {
             PreparedQuery results = dataStore.prepare(query);
             Entity resultsEntity = results.asSingleEntity();
             if (resultsEntity == null) {
+                // Query for the name
```

Comment:
Nit: This is not something you absolutely need to change right now, but to better decouple things, it'd be best to depend on RestaurantClient and call its method, so you only implement the query in one place.